core/primitives/src/shard_layout.rs

```diff
@@ -104,17 +104,19 @@ impl ShardLayout {
     }
 
     #[inline]
-    pub fn get_split_shards(&self, parent_shard_id: ShardId) -> Option<Vec<ShardUId>> {
+    pub fn get_split_shard_uids(&self, parent_shard_id: ShardId) -> Option<Vec<ShardUId>> {
+        self.get_split_shard_ids(parent_shard_id).map(|shards| {
+            shards.into_iter().map(|id| ShardUId::from_shard_id_and_layout(id, &self)).collect()
+        })
+    }
+
+    #[inline]
+    pub fn get_split_shard_ids(&self, parent_shard_id: ShardId) -> Option<Vec<ShardId>> {
         match self {
             Self::V0(_) => None,
             Self::V1(v1) => match &v1.shards_split_map {
                 Some(shards_split_map) => match shards_split_map.get(parent_shard_id as usize) {
```

Comment:
no need for this match now: `=> hards_split_map.get(parent_shard_id as usize).cloneds()`